stumpy/stumpi.py

```diff
@@ -106,34 +117,41 @@ def __init__(self, T, m, egress=True, normalize=True, p=2.0):
         p : float, default 2.0
             The p-norm to apply for computing the Minkowski distance. This parameter is
             ignored when `normalize == True`.
+
+        k : int, default 1
+            The number of top `k` smallest distances used to construct the matrix
+            profile. Note that this will increase the total computational time and
+            memory usage when `k > 1`.
         """
         self._T = core._preprocess(T)
         core.check_window_size(m, max_size=self._T.shape[-1])
         self._m = m
+        self._k = k
+
         self._n = self._T.shape[0]
         self._excl_zone = int(np.ceil(self._m / config.STUMPY_EXCL_ZONE_DENOM))
         self._T_isfinite = np.isfinite(self._T)
         self._egress = egress
 
-        mp = stump(self._T, self._m)
-        self._P = mp[:, 0].astype(np.float64)
-        self._I = mp[:, 1].astype(np.int64)
-        self._left_I = mp[:, 2].astype(np.int64)
-        self._left_P = np.empty(self._P.shape, dtype=np.float64)
-        self._left_P[:] = np.inf
+        mp = stump(self._T, self._m, k=self._k)
+        self._P = mp[:, :k].astype(np.float64)
+        self._I = mp[:, k : 2 * k].astype(np.int64)
+
+        self._left_I = mp[:, 2 * k].astype(np.int64)
+        self._left_P = np.full_like(self._left_I, np.inf, dtype=np.float64)
 
         self._T, self._M_T, self._Σ_T = core.preprocess(self._T, self._m)
         # Retrieve the left matrix profile values
 
-        # Since each matrix profile value is the minimum between the left and right
-        # matrix profile values, we can save time by re-computing only the left matrix
-        # profile value when the matrix profile index is equal to the right matrix
-        # profile index.
-        mask = self._left_I == self._I
-        self._left_P[mask] = self._P[mask]
+        # Since each (top-1) matrix profile value is the minimum between the left
+        # and right matrix profile values, we can save time by re-computing only
+        # the left matrix profile value when the (top-1) matrix profile index is
+        # equal to the right matrix profile index.
```

Comment:
right after this sentence where we read `... equal to the right matrix profile index`, we see this line: `mask = self._left_I == self._I[:, 0]`. 

Do you think it is confusing?